### ISSUE TITLE
Remove empty query string.

### ISSUE THREAD

**Issue by sideshot** [paoloricciuti/sveltekit-search-params#78]
### Describe the bug

How do I get an empty query string off the URL?  

### Reproduction

Add with data then try to remove.

**Comment by paoloricciuti** [comment]
You can set the empty string as default value and use showDefaults: false

**Comment by sideshot** [comment]
Strange, that is what I'm doing.  I saw that suggestion in another issue.  Once I select and then deselect, it still ends up with ?filter_category=


```
	const selected = queryParam(`filter_${section.id}`,ssp.string(''),{
		showDefaults: false
	});
```
```
<input
	id={`filter-${section.id}-${optionIdx}`}
	value="{option.value}"
	type="checkbox"
	bind:group={$selected}
	class="h-4 w-4 rounded border-gray-300 text-blue-600 focus:ring-blue-500"
/>
```

Any thoughts?  Awesome script, by the way.

**Comment by paoloricciuti** [comment]
I don't think you can use a string to bind to a checkbox group...if you prepare a reproduction I can try to take a look

**Comment by herronelou** [comment]
It seems similar to issue #72, at first the defaults don't show in the URL, but upon setting them and reverting them, they stay in the URL.

```
<script lang="ts">

    import { queryParam, queryParameters, ssp } from 'sveltekit-search-params';

    const test = queryParam('search', ssp.string(''), { debounceHistory: 500, showDefaults: false});

</script>

<input type="text" bind:value={$test}>
```

Before:
![image](https://github.com/user-attachments/assets/8f826861-f259-4164-bb0f-e417c7f4110b)

After typing something:
![image](https://github.com/user-attachments/assets/f25d0567-d948-4ff2-b2ad-44c4e24c4435)


After deleting it:
![image](https://github.com/user-attachments/assets/89f7be7b-4c7a-48b6-86b2-d3d435d22d20)

**Comment by ken-zlai** [comment]
Try setting the value to `null` instead of an empty string

**Comment by gwax** [comment]
I think this is a regression. Certainly, in the `next` branch, we navigate to "value" even if value is equal to defaultValue.

In the proxy creation: https://github.com/paoloricciuti/sveltekit-search-params/blob/next/src/lib/sveltekit-search-params.svelte.ts#L266

I believe the do_navigate should be modified to something of the form:
```js
do_navigate(
  name,
  valueIsDefault ? undefined : value,
  encodes,
  overrides,
  navigation_options,
);
```

but I'm not yet clear on the best way to feed through a notion of `valueIsDefault`

**Comment by paoloricciuti** [comment]
I don't think it's a regression, this was an issue when next wasn't even a thing

**Comment by gwax** [comment]
Sorry, it worked in the svelte4 version at one point, and then it regressed. I can't put my finger on exactly when but I can say for certain that the behavior of `ssp.string("")` is currently to leave an empty string in the query parameters for both svelte4 and next.
